src/common_rl/from_env.jl

```diff
@@ -0,0 +1,215 @@
+abstract type AbstractRLEnvMDP{S, A} <: MDP{S, A} end
+abstract type AbstractRLEnvPOMDP{S, A, O} <: POMDP{S, A, O} end
+
+const AbstractRLEnvProblem = Union{AbstractRLEnvMDP, AbstractRLEnvPOMDP}
+
+POMDPs.actions(m::AbstractRLEnvProblem) = RL.actions(m.env)
+POMDPs.discount(m::AbstractRLEnvProblem) = m.discount
+
+function POMDPs.actions(m::AbstractRLEnvProblem, s)
+    if provided(RL.valid_actions, m.env)
+        RL.setstate!(m.env, s)
+        return RL.valid_actions(m.env)
+    else
+        return RL.actions(m.env)
+    end
+end
+
+function POMDPs.initialstate(m::AbstractRLEnvProblem)
+    return ImplicitDistribution(m) do m, rng
+        RL.reset!(m.env)
+        return RL.state(m.env)
+    end
+end
+
+function POMDPs.isterminal(m::AbstractRLEnvProblem, s)
+    RL.setstate!(m.env, s)
+    return RL.terminated(m.env)
```

Comment:
Same as above. What if we just want to check if a particular state is terminal without changing the environment necessarily.